hackerone_alchemy.py

```diff
@@ -179,18 +180,32 @@ def word_in_report(self, report, word):
         return any(word in field.lower() for field in text_fields)
 
     def get_bonus_information(self, reports):
-        resolved_by_reporter = collections.defaultdict(list)
-        return_dict = {}
+        # Assumes `reports` is in reverse chronological order by creation date
+        accepted_by_reporter = collections.defaultdict(list)
 
         for report in reports:
-            if report.state == "resolved":
-                resolved_by_reporter[report.reporter.username] = report
+            if report.state in {"resolved", "triaged"}:
+                accepted_by_reporter[report.reporter].append(report)
 
-        for reporter, reports_by_reporter in resolved_by_reporter.items():
+        reporter_rewards = {}
+        for reporter, reports_by_reporter in accepted_by_reporter.items():
             if len(reports_by_reporter) >= 5:
-                return_dict[reporter] = reports_by_reporter
+                reporter_rewards[reporter] = self.calc_report_bonuses(reports_by_reporter)
 
-        return return_dict
+        return reporter_rewards
+
+    def calc_report_bonuses(self, reports):
+        report_bonuses = {}
+
+        # The first four reports are not eligible
+        eligible_reports = reports[:-4]
+
+        for report in eligible_reports:
+            other_reports = [r for r in reports if r is not report]
+            avg_bounty = sum(r.total_bounty for r in other_reports) / len(other_reports)
```

Comment:
Nope, this should never be called with less than 5 reports, and `eligible_reports` will be empty and the `for` will never fire if there's less (due to the `eligible_reports = reports[:-4]` above.)